ddpui/api/public_api.py

```
@@ -1070,6 +1072,50 @@ def get_public_chart_data_preview_total_rows(request, token: str, chart_id: int)
+    try:
+        result = KPIService.get_kpi_data(
+            kpi_id,
+            dashboard.org,
+            time_grain_override=time_grain,
+            date_from=date_from,
+            date_to=date_to,
+            dashboard_filters=parsed_dashboard_filters,
+        )
+    except KPINotFoundError:
+        return 404, PublicErrorResponse(error="KPI not found", is_valid=False)
+    except Exception as e:
```

Ruff (0.15.21)

[warning] Do not catch blind exception: `Exception`

(BLE001)